### ISSUE TITLE
Setting to enable/disable Cache

### ISSUE THREAD

**Issue by lowedown** [mikeedwards83/Glass.Mapper#166]
Hi

It would be great if it you could add a possibility to enable/disable caching via config. Currently it is only possible to enable/disable it via Annotations in code.
We need this because we only want to use caching on our delivery instances and disable it on authoring instances.

**Comment by cardinal252** [comment]
You can use the DisableCache class to disable it globally as far as I know, either by using:

DisableCache disableCache = new DisableCache();

or

DisableCache.Current = CacheSetting.Disabled;

I would recommend the first as its the official way of doing it.

Simply patch in a pipeline processor to your environment and this should do the trick.

**Comment by lowedown** [comment]
Thanks fo your help.
I added a setting to Glass.Mapper.Sc.config
```
<setting name="Glass.Mapper.Sc.Caching.Enabled" value="true"/>
```

And the following code to Start() in GlassMapperSc.cs:.
```
if (!Settings.GetBoolSetting("Glass.Mapper.Sc.Caching.Enabled", false))
{
    var disableCache = new DisableCache();
}
```

That way caching can be switched on/off using config. Of course the models still need the Cachable=true annotation in the model.

**Comment by lowedown** [comment]
Caution: this will only work if there is no 
```
using (new DisableCache()) { ... }
```
anywhere in your code. The Dispose() method of DisableCache() will re-enable caching!

Therefore it would still be nice if we had another global way of disabling caching.

**Comment by mikeedwards83** [comment]
Released in 4.0.4
